modules/exploits/windows/mssql/mssql_payload.rb

```diff
@@ -68,7 +70,11 @@ def initialize(info = {})
   end
 
   def check
-    if !mssql_login_datastore
+    if session
```

Comment:
Same as above there are a few instances were we check if we have a `session` and then check it again while checking `mssql_login_datastore`. Maybe this logic can be combined.

Similar to [here](https://github.com/rapid7/metasploit-framework/pull/18759/files#diff-d0f7dcac42dabfdf1e74496e6d9749179d9cbab739896c6a28786a6cff0660b2R108-R116).